src/content/documentation/modifiers.md

```diff
@@ -3,12 +3,242 @@ title: Modifiers
+
+The `small` modifier will apply to `button`, while `modal` and `with a header`
+will apply to `dialog`. This is because `in a...` has changed the modifier
+target to `dialog`. Now you can target a very specific variant of a `dialog` in
+your `button` styles if you need to.
+
+```scss
+@include restyle-define(button, (
+  ...default styles...
+
+  '@restyle.modifiers': (
+    small: (
+      ...
+    ),
+    ...
+    'in a modal dialog with a header': (
+      display: block
+    )
+  )
+));
+```
+
+Additionally, reSTYLE is flexible enough to match a more generic context
+modifier from your pattern definition if you don't provide one as specific as
+what was requested. So our example of `in a modal dialog with a header` in the
+`restyle()` include call could also match `in a modal dialog` or just `in a
+dialog` from a pattern definition.
+
+If you provide multiple modifier definitions that match a context modifier,
+reSTYLE will merge their styles, with the last definition defined winning when
+multiple definitions define the same property.
+
+```scss
+@include restyle-define(button, (
+  ...default styles...
+
+  '@restyle.modifiers': (
+    // All of these will match for 'in a modal dialog with a header'
+    'in a dialog': (
+      ...
+      color: #000
+    ),
+    'in a modal dialog': (
+      ...
+      color: #fff
+    ),
+    'in a modal dialog with a header': (
+      ...
+      color: #333 // This `color` wins.
+    )
+  )
+));
+```
+
+## Composability
+
+You can compose modifiers in your `restyle()` calls to produce more pattern
+style variations. Composition of variant modifiers works the same way as it does
+with context modifiers. Let's say you have a modifier definition for `primary
+button` and one for `small button`. You can ask `restyle()` for a `small primary
+button` when invoking your `button` pattern. You can think of `small primary` as
+a **Compound Modifier**.
+
+If you don't provide a modifier definition for `small primary` but do define
+`small` and `primary` separately, reSTYLE will figure out the styles for you by
+merging the two style definitions in a top-down, last definition taking
+precedence approach.
+
+```scss
+@include restyle-define(button, (
+  ...default styles...
+
+  '@restyle.modifiers': (
+    small: (
+      font-size: 90%,
+      font-weight: normal,
+      padding: 3px 7px
+    ),
+    primary: (
+      background-color: #0033cc,
+      border-color: #0033cc,
+      color: #fff,
+      font-weight: bold
+    )
+  )
+));
+```
+
+You can also specify specific styles that you want to apply for a compound
+modifier.
+
+```scss
+@include restyle-define(button, (
+  ...default styles...
+
+  '@restyle.modifiers': (
+    small: (
+      ...
+    ),
+    primary: (
+      ...
+    ),
+    'small primary': (
+      font-size: 95%
+    )
+  )
+));
+```
+
+Between compound modifiers and context modifiers, reSTYLE empowers you to
+compose your UI elements using mixtures of variations and to compose larger
```

Comment:
`UI elements` -> `UI patterns`

`mixtures of variations` -> `flexible language`